tools/crm_mon.c

```diff
@@ -988,48 +988,60 @@ pacemakerd_status(void)
-                            */
-                            print_as(output_format ,"Pacemaker daemons shut down - reporting to SBD ...\n");
-                            break;
-                        default:
-                            break;
-                    }
-                } else {
-                    switch (state) {
-                        case pcmk_pacemakerd_state_running:
-                            rc = pcmk_rc_ok;
-                            break;
-                        default:
-                            break;
+                rc = pcmk_poll_ipc(pacemakerd_api, options.reconnect_msec/2);
+                if (rc == pcmk_rc_ok) {
+                    pcmk_dispatch_ipc(pacemakerd_api);
+                    rc = ENOTCONN;
+                    if ((output_format == mon_output_console) ||
+                        (output_format == mon_output_plain)) {
+                        switch (state) {
+                            case pcmk_pacemakerd_state_running:
+                                rc = pcmk_rc_ok;
+                                break;
+                            case pcmk_pacemakerd_state_starting_daemons:
+                                print_as(output_format ,"Pacemaker daemons starting ...\n");
+                                break;
+                            case pcmk_pacemakerd_state_wait_for_ping:
+                                print_as(output_format ,"Waiting for startup-trigger from SBD ...\n");
+                                break;
+                            case pcmk_pacemakerd_state_shutting_down:
+                                print_as(output_format ,"Pacemaker daemons shutting down ...\n");
+                                break;
+                            case pcmk_pacemakerd_state_shutdown_complete:
+                                /* assuming pacemakerd doesn't dispatch any pings after entering
+                                * that state unless it is waiting for SBD
+                                */
+                                print_as(output_format ,"Pacemaker daemons shut down - reporting to SBD ...\n");
+                                break;
+                            default:
+                                break;
+                        }
+                    } else {
+                        switch (state) {
+                            case pcmk_pacemakerd_state_running:
+                                rc = pcmk_rc_ok;
+                                break;
+                            default:
+                                break;
+                        }
                     }
                 }
             }
-        }
+            break;
+        case EREMOTEIO:
+            rc = pcmk_rc_ok;
+            if ((output_format == mon_output_console) ||
+                        (output_format == mon_output_plain)) {
+                print_as(output_format, "Running on remote-node ...\n");
```

Comment:
I think we can skip a message here to avoid clutter, since we're not actually doing anything

Reply:
hmm ... actually was quite nice in my tests as I could see the difference between nothing running at all and cluster not connected to pacemaker-remote.

Reply:
maybe limit it to curses-console ...

Reply:
Sounds good, let's limit it to console and reword it to something like "Waiting for cluster connection to this remote node" or whatnot. For text, maybe we use a similar message for the error output (maybe via the GError).

Reply:
flag on_remote_node is clumsy but I guess with the overhaul that makes print_as go away this can as well go again.